### PR TITLE
Add sanity test for Sqlite and add purge table.

### PR DESCRIPTION
**Pull request by kberg**
(but it doesn't work, gah)

### CODE REVIEW

**Review by bafolts — APPROVED** [review]

**Comment by kberg** [comment]
@bafolts take another look, please. I added another test. Am not going to change cleanSaves now. But I am testing the code I added, and how about that? It didn't actually work! :D

File: tests/database/SQLite.spec.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  constructor() {
+    super(IN_MEMORY_SQLITE_PATH, true);
+  }
+
+  public get database() {
+    return this.db;
+  }
+
+  public async saveGame(game: Game) {
+    this.saveGamePromise = super.saveGame(game);
```
Comment (bafolts) [inline tests/database/SQLite.spec.ts:20]: YES! As I wrote my version in javascript I realized it could be written this way! Awesome.

Comment (kberg) [reply tests/database/SQLite.spec.ts:20]: Look at the failed test, though.

**Comment by kberg** [comment]
This is a surprising failed test which will make things flaky. Not sure what to do about this.

```
  1) SQLite
       saveIds:

      AssertionError: expected [ 0, 2, 3 ] to have the same members as [ 0, 1, 2, 3 ]
      + expected - actual

       [
         0
      +  1
         2
         3
       ]
      
      at /home/runner/work/terraforming-mars/terraforming-mars/tests/database/SQLite.spec.ts:84:28
      at step (tests/database/SQLite.spec.ts:48:23)
      at Object.next (tests/database/SQLite.spec.ts:29:53)
      at fulfilled (tests/database/SQLite.spec.ts:20:58)
```

File: tests/database/SQLite.spec.ts (hunk 1 above)
Comment (kberg) [reply tests/database/SQLite.spec.ts:20]: Looks much better. Thank you.

**Review by bafolts — APPROVED** [review]
This is good change. We need to keep bubbling this change upward and having functions return a `Promise` that doesn't resolve until we have created this save entry. Right now a lot of functions return and leave a `Promise` that we have no pointer to that resolves at some point later.

**Comment by kberg** [comment]
Thanks for making it work! I fully expect it to start failing. :( flaky tests.

**Pull request merged**